### ISSUE TITLE
UL management

### ISSUE THREAD

**Issue by bastien-solutions** [suyashkumar/dicom#136]
Hello @suyashkumar ,

First of all, thank you for your very useful lib (there is a lack in Go on this domain).
I have a question about roadmap, did you plan to manage upper layer network protocol ?

Regards,

B.

**Comment by protheusfr** [comment]
+1 this could really make this dice lib as an alternative to dcmtk or dcm4che.

**Comment by suyashkumar** [comment]
Thanks for the suggestion and the kind words! I've been thinking about putting something like this on the roadmap (maybe more offering a dicomWEB client and/or server package that work well with this library). 

If possible, can you share a little more info about both of your usecases, and what functionality in particular you would find most valuable?

**Comment by bastien-solutions** [comment]
Of course, as you know the DICOM standard does not only specify a file format but also a network protocol.

The latter is used, for example, by all medical imaging devices (modalities), PACs, viewers, analysis consoles, etc. to communicate with each other.

The WEB version (DICOMWeb / WADO) has been emerging for a few years mainly for Web uses (as its name indicates), but 99% of medical data exchanges still use the old DICOM UL protocol.

Today, mainly two Open source libraries implement it: DCMTK (C++) and DCM4CHE (Java). 
They are both of a certain age (20 years old for DCM4CHE, 24 years old for DCMTK), their maintenance and documentation is becoming more and more difficult, and I think there is room for a new generation of OpenSource libraries covering the entire standard.

It would allow the emergence of a new generation of medical software, more modern, more secure, more "Cloud native" and the Go language is perfect for that.

With the advent of digitized patient records, AI, graphical oriented databases, and access to healthcare for the largest number of people, it may be time to dust off the DICOM environment.

I can provide some of my time for contribution.
